### PR TITLE
feat: Rethink phishing detector bot

### PR DESCRIPTION
**Pull request by dgusakov**

### CODE REVIEW

File: phishing-detect/src/agent-phishing-detect.spec.ts [hunk 1]
```diff
@@ -74,32 +74,36 @@ describe("phishing-detect", () => {
...
         new Set(
           Array.from({ length: UNIQ_DELEGATES_THRESHOLD_CONTRACT - 1 }, () =>
-            randromAddress()
+            randomAddress()
           )
         )
       );
 
-    const findings = await handleTransaction(txEventWithHighApproval);
+    handleTransaction(txEventWithHighApproval);
```
Comment (madlabman) [inline phishing-detect/src/agent-phishing-detect.spec.ts:82]: It's better to extract to the separate test case, which will check state changes only.

File: phishing-detect/src/agent-phishing-detect.ts [hunk 2]
```diff
@@ -59,250 +36,101 @@ export async function initialize(
...
+  await Promise.all(
+    Array.from(spenders.keys()).map(async (spender) => {
+      let spenderInfo = spenders.get(spender);
+      if (!spenderInfo) {
+        return;
+      }
+      if (spenderInfo.isContract == undefined) {
+        spenderInfo.isContract = await isContract(spender);
+      }
+      const uniqTokenTypes = spenderInfo.tokens.size;
```
Comment (madlabman) [inline phishing-detect/src/agent-phishing-detect.ts:103]: Unused?

File: phishing-detect/src/agent-phishing-detect.spec.ts (hunk 1 above)
Comment (dgusakov) [reply phishing-detect/src/agent-phishing-detect.spec.ts:82]: Why? This is a common flow. TX -> Block

Comment (madlabman) [reply phishing-detect/src/agent-phishing-detect.spec.ts:82]: It's a unit test. And you see, there are two things under test.

Comment (dgusakov) [reply phishing-detect/src/agent-phishing-detect.spec.ts:82]: Done

**Review by madlabman — APPROVED** [review]

**Pull request merged**